src/peft/tuners/tuners_utils.py

```diff
@@ -914,6 +914,20 @@ def check_target_module_exists(config, key: str) -> bool | re.Match[str] | None:
         `bool` | `re.Match[str]` | `None`: True of match object if key matches any target modules from config, False or
         None if no match found
     """
+    if config.target_modules and config.exclude_modules:
+        if config.exclude_modules == config.target_modules:
```

Comment:
I don't think that this is a sufficient check. Just to give some examples:

- `target_modules` is a list of str and `exclude_modules` is a str, but both essentially target the same modules
- `target_modules` is `["lin0", "lin1", "lin2"]` and `exclude_modules` is `["lin0", "lin1"]`. They look different, but if `"lin2"` doesn't exist, they're actually the same

We thus need to check what modules were actually targeted, we can't rely on the passed arguments.